### PR TITLE
feat: support .NET

### PR DESCRIPTION
**Pull request by olljanat**
Targets to support .NET "publish" and "test" commands on same way like Golang build and test works.

~Marked as draft because unit test(s) to verify "test" command is not yet implemented.~

### CODE REVIEW

**Comment by gerhard** [comment]
Thank you @olljanat for your first contribution! We just kicked off the workflow for you 👟

Would love to get your 👀 & 🙇‍♂️ on this one @cdhunt & @nabsul.

Should this be added to the experimental namespace - `universe.dagger.io/x/<MAINTAINER_EMAIL>/dotnet` - @shykes, same as #1991? If yes, then we should add this to https://github.com/dagger/.github/blob/main/CONTRIBUTING.md

File: pkg/universe.dagger.io/dotnet/build.cue [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package dotnet
+
+import (
+	"dagger.io/dagger"
+)
+
+// Build a dotnet binary
+#Build: {
```
Comment (cdhunt) [inline pkg/universe.dagger.io/dotnet/build.cue:8]: Naming the action `#Build` when it calls `publish` could be confusing. I know you're following the existing Dagger convention, but my suggestion is to mimic the dotnet SDK command names.

**Comment by olljanat** [comment]
Moved to experimental namespace, renamed `#Build` command to ´#Publish`, fixed cuelint issues and added support for `#Test` command. This should be ready for merge unless you find other issues from it.

File: pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+// Publish a dotnet binary
+#Publish: {
+	// Source code
+	source: dagger.#FS
+
+	// Target package to publish
+	package: *"." | string
+
+	env: [string]: string
```
[the remark this answers is not in the document]
Comment (olljanat) [reply pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue:15]: Probably left over as I start from Go package. Will drop it as `dotnet publish` does not actually care about environment variables (afaik).

[the remark this answers is not in the document]
Comment (cdhunt) [reply pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue]: `dotnet publish` doesn't, but MSBuild can/does and dotnet might call out to msbuild depending on the project configuration.

[the remark this answers is not in the document]
Comment (olljanat) [reply pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue:15]: Valid point. Now environment variables should be available inside of container too.

**Review by aluzzardi — APPROVED** [review]
Thank for you for the contribution, @olljanat!

Looks good to merge after CI passes.

File: pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+	container: #Container & {
+		"source": source
+		"env": {
```
Comment (TomChv) [inline pkg/universe.dagger.io/x/olli.janatuinen@gmail.com/dotnet/publish.cue:19]: `"env": env` is enough ;)

**Review by TomChv — APPROVED** [review]
LGTM too, I left a style comment but it's mergeable after CI passes

**Pull request merged**